### ISSUE TITLE
Databricks dialect does not properly support column_properties for `COLLATE` syntax

### ISSUE THREAD

**Issue by krajewf** [sqlfluff/sqlfluff#7747]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff fails with unparsable section error when create table column properties contains `COLLATE` clause

### Expected Behaviour

sqlfluff parses COLLATE as a valid databricks table column properties clause

### Observed Behaviour

Databricks [COLLATE (TABLE)](https://docs.databricks.com/aws/en/sql/language-manual/sql-ref-syntax-ddl-create-table-using#syntax) documentation shows COLLATE as valid syntax, thus sqlfluff databricks dialect should parse COLLATE syntax

### How to reproduce

test-collate.sql
```sql
CREATE TABLE test (
  id STRING COLLATE UTF8_BINARY NOT NULL COMMENT 'Required ID field'
)
USING DELTA
```

`sqlfluff lint --dialect databricks test-collate.sql`

```bash
== [test-collate.sql] FAIL
L:   1 | P:  19 |  PRS | Line 1, Position 19: Found unparsable section: '(\n  id
                       | STRING COLLATE UTF8_BINARY NOT NU...'
WARNING: Parsing errors found and dialect is set to 'databricks'. Have you configured your dialect correctly?
All Finished!
```

test-no-collate.sql
```sql
CREATE TABLE test (
  id STRING NOT NULL COMMENT 'Required ID field'
) USING DELTA
```

``sqlfluff lint --dialect databricks test-collate.sql`

```bash
All Finished 📜 🎉!
```

### Dialect

databricks

### Version

sqlfluff, version 4.1.0

### Configuration

```
sqlfluff]
dialect = databricks
max_line_length = 0
large_file_skip_byte_limit = 0
rules = CP01

[sqlfluff:indentation]
tab_space_size = 2

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:rules:capitalisation.identifiers]
extended_capitalisation_policy = lower
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
